docs/book/src/tasks/experimental-features/cluster-class/write-clusterclass.md

```diff
@@ -1,3 +1,618 @@
+
+</aside>
+
+<aside class="note">
+
+<h1>Tip: clusterctl alpha topology plan</h1>
+
+The `clusterctl alpha topology plan` command can be used to test ClusterClasses; the output will show
+you how the resulting Cluster will look like, but without actually creating it.
+For more details please see: [clusterctl alpha topology plan].
+
+</aside>
+
+## ClusterClass with MachineHealthChecks
+
+`MachineHealthChecks` can be configured in the ClusterClass for the control plane and for a 
+MachineDeployment class. The following configuration makes sure a `MachineHealthCheck` is 
+created for the control plane and for every `MachineDeployment` using the `default-worker` class.
+
+```yaml
+apiVersion: cluster.x-k8s.io/v1beta1
+kind: ClusterClass
+metadata:
+  name: docker-clusterclass-v0.1.0
+spec:
+  controlPlane:
+    ...
+    machineHealthCheck:
+      maxUnhealthy: 33%
+      nodeStartupTimeout: 15m
+      unhealthyConditions:
+      - type: Ready
+        status: Unknown
+        timeout: 300s
+      - type: Ready
+        status: "False"
+        timeout: 300s
+  workers:
+    machineDeployments:
+    - class: default-worker
+      ...
+      machineHealthCheck:
+        unhealthyRange: "[0-2]"
+        nodeStartupTimeout: 10m
+        unhealthyConditions:
+        - type: Ready
+          status: Unknown
+          timeout: 300s
+        - type: Ready
+          status: "False"
+          timeout: 300s
+```
+
+## ClusterClass with patches
+
+As shown above, basic ClusterClasses are already very powerful. But there are cases where 
+more powerful mechanisms are required. Let's assume you want to manage multiple Clusters 
+with the same ClusterClass, but they require different values for a field in one of the 
+referenced templates of a ClusterClass. A concrete example would be to deploy Clusters in 
+different AWS regions. In this case, every cluster needs a Cluster-specific value for 
+`.spec.region` in `AWSCluster`. Use cases like this can be implemented with ClusterClass 
+patches.
+
+**Defining variables in the ClusterClass**
+
+The following example shows how variables can be defined in the ClusterClass.
+A variable definition specifies the name and the schema of a variable and if it is 
+required. The schema defines how a variable is defaulted and validated. It supports 
+a subset of the schema of CRDs. For more information please see the [godoc](https://doc.crds.dev/github.com/kubernetes-sigs/cluster-api/cluster.x-k8s.io/ClusterClass/v1beta1#spec-variables-schema-openAPIV3Schema).
+
+```yaml
+apiVersion: cluster.x-k8s.io/v1beta1
+kind: ClusterClass
+metadata:
+  name: docker-clusterclass-v0.1.0
+spec:
+  ...
+  variables:
+  - name: imageRepository
+    required: true
+    schema:
+      openAPIV3Schema:
+        type: string
+        description: ImageRepository is the container registry to pull images from.
+        default: k8s.gcr.io
+        example: k8s.gcr.io
+```
+
+<aside class="note">
```

Comment:
```suggestion
<aside class="note">
<h1>Supported types</h1>
```
I find that having a title in the note helps readability

Reply:
Good point!